dis_003_volcano_reports/contents/src/__init__.py

```diff
@@ -30,184 +38,317 @@
+            max_rows: maximum rows that can be stored in the Carto table (integer)
+            time_field: column that stores datetime information (string) 
+            max_age: oldest date that can be stored in the Carto table (datetime object)
+    RETURN  num_dropped: number of rows that have been dropped from the table (integer)
+    ''' 
+    # initialize number of rows that will be dropped as 0
     num_dropped = 0
+
+    # check if max_age is a datetime object
     if isinstance(max_age, datetime.datetime):
+        # convert max_age to a string 
         max_age = max_age.isoformat()
 
-    # 1. delete by age
+    # if the max_age variable exists
     if max_age:
-        r = cartosql.deleteRows(table, "{} < '{}'".format(time_field, max_age))
+        # delete rows from table which are older than the max_age
+        r = cartosql.deleteRows(table, "{} < '{}'".format(time_field, max_age), user=CARTO_USER, key=CARTO_KEY)
+        # get the number of rows that were dropped from the table
         num_dropped = r.json()['total_rows']
 
-    # 2. get sorted ids (old->new)
-    r = cartosql.getFields('cartodb_id', table, order='{}'.format(time_field),
-                           f='csv')
+    # get cartodb_ids from carto table sorted by date (new->old)
+    r = cartosql.getFields('cartodb_id', table, order='{} desc'.format(time_field),
+                           f='csv', user=CARTO_USER, key=CARTO_KEY)
+    # turn response into a list of strings of the ids
     ids = r.text.split('\r\n')[1:-1]
 
-    # 3. delete excess
+    # if number of rows is greater than max_rows, delete excess rows
     if len(ids) > max_rows:
-        r = cartosql.deleteRowsByIDs(table, ids[:-max_rows])
+        r = cartosql.deleteRowsByIDs(table, ids[max_rows:], user=CARTO_USER, key=CARTO_KEY)
+        # get the number of rows that have been dropped from the table
         num_dropped += r.json()['total_rows']
     if num_dropped:
         logging.info('Dropped {} old rows from {}'.format(num_dropped, table))
-    return num_dropped
 
-###
-## Accessing remote data
-###
+    return(num_dropped)
 
 def genUID(lat,lon,dt):
+    '''Generate unique id using latitude, longitude and date information
+    INPUT   lat: latitude of the earthquake (float)
+            lon: longitude of the earthquake (float)
+            dt: date for which we want to generate id (string)
+    '''
     return '{}_{}_{}'.format(lat,lon,dt)
 
-def processData(existing_ids):
-    """
-    Inputs: FTP SOURCE_URL and filename where data is stored, existing_ids not to duplicate
-    Actions: Retrives data, dedupes and formats it, and adds to Carto table
-    Output: Number of new rows added
-    """
+def processData(url, existing_ids):
+    '''
+    Fetch, process and upload new data
+    INPUT   url: url where you can find the source data (string)
+            existing_ids: list of date IDs that we already have in our Carto table (list of strings)
```

Comment:
removing the word date because the uid is made from lat, lon, and date
```suggestion
            existing_ids: list of IDs that we already have in our Carto table (list of strings)
```